.github/workflows/test-changed-firestore.yml

```diff
@@ -35,23 +35,27 @@ jobs:
           yarn
       - name: build
         id: build
-        # TODO(wuandy): Separate yarn and egrep into steps, so build failure
-        # is captured by github actions.
-        run: yarn build:changed firestore | egrep "Skipping all"
+        run: |
+          set -o pipefail
+          yarn build:changed firestore | tee ${{ runner.temp }}/yarn.log.txt
+        continue-on-error: false
+      - name: Check if Firestore is changed
+        id: check-changed
+        run: egrep "Skipping all" ${{ runner.temp }}/yarn.log.txt
         # Continue when "Skipping all" is not found
         continue-on-error: true
       - name: set output
         # This means "Skipping all" was not found
-        if: steps.build.outcome != 'success'
+        if: steps.check-changed.outcome != 'success'
```

Comment:
I don't really like that failure of the `egrep` command is conflated with the absence of "skipping all" from the output. Is there a more robust way to test for skipping the tests that you can think of?

Update: What about something like this:

```
skip_count=$(grep --count "Skipping all" test.txt)
echo "skip_count=$skip_count" >> "$GITHUB_OUTPUT"
```

Then later on you can do something like this:

```
if: steps.check-changed.outputs.skip_count == 0
```

Reply:
I think it's more worthwhile to change the script give more explicit result than doing grepping at all...not today though..i had enough github actions work recently.